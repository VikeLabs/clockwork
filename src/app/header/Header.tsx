import { Center, Grid, GridItem, Flex, Text, LinkBox } from '@chakra-ui/react';
import { Link } from 'react-router-dom';

import { NavButtons } from './components/NavButtons';
import { Search } from './components/SearchBar';
import { TermButtons } from './components/TermButtons';

export interface HeaderProps {
  onSearchChange?: (query: string) => void;
  onSearchSubmit?: (query: string) => void;
}
/**
 * Primary UI component for content
 */
export function Header({ onSearchChange }: HeaderProps): JSX.Element {
  return (
    <Grid
      templateColumns="repeat(3, 1fr)"
      as="header"
      py="1.5"
      px="8"
      boxShadow="md"
      bg="#82cbee"
      zIndex={100}
      maxH="56px"
    >
      <GridItem colSpan={1}>
        <Flex justifyContent="space-between" alignContent="center" alignItems="flex-start">
          <LinkBox as={Link} to="/" bg="transparent" border="none" ml={5}>
            <Text fontSize="xl" fontWeight="bold" color="white">
              CourseUp
            </Text>
          </LinkBox>
          <NavButtons />
        </Flex>
      </GridItem>
      <GridItem colStart={2}>
        <Center>
          <Search onChange={onSearchChange} />
        </Center>
      </GridItem>
      <GridItem colStart={3}>
        <TermButtons />
      </GridItem>
    </Grid>
  );
}
