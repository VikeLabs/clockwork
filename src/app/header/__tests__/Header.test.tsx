import { render } from '@testing-library/react';

import { Header } from '../Header';

describe('Header', () => {
  it('renders the correct content', () => {
    expect(render(<Header />).container).toMatchSnapshot();
  });
});
