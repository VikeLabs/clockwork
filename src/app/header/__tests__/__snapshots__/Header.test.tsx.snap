// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the correct content 1`] = `
<div>
  <header
    class="css-1qfmfr6"
  >
    <div
      class="css-gmuwbf"
    >
      <p
        class="chakra-text css-8b95vw"
      >
        clockwork
      </p>
    </div>
    <div
      class="css-17xejub"
    />
    <div
      class="chakra-stack css-b2vdeo"
    >
      <div
        class="chakra-input__group css-4302v8"
      >
        <input
          class="chakra-input css-hzknmz"
          placeholder="Search a course"
          value=""
        />
        <div
          class="chakra-input__right-addon css-mqmw1z"
        >
          <button
            aria-label="Search"
            class="chakra-button css-sd7an6"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,15 L22,22 L15,15 Z M9.5,17 C13.6421356,17 17,13.6421356 17,9.5 C17,5.35786438 13.6421356,2 9.5,2 C5.35786438,2 2,5.35786438 2,9.5 C2,13.6421356 5.35786438,17 9.5,17 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="chakra-button__group css-1vr4fdn"
        role="group"
      >
        <button
          class="chakra-button css-1rxajhb"
          name="202009"
          type="button"
        >
          Fall 2020
        </button>
        <button
          class="chakra-button css-1rxajhb"
          name="202101"
          type="button"
        >
          Spring 2021
        </button>
        <button
          class="chakra-button css-1rxajhb"
          name="202105"
          type="button"
        >
          Summer 2021
        </button>
      </div>
    </div>
    <div
      class="css-17xejub"
    />
  </header>
</div>
`;
